### DISCUSSION TITLE
TypeError: Right-hand side of 'instanceof' is not an object

### DISCUSSION THREAD

**Discussion by theDanielJLewis** [typegoose/typegoose#836] [Q&A]
I'm pulling my hair out trying to solve this but I can't.

Simplified source:

```js
export class PodcastClass {
	@prop()
	public oldId?: number;
}

export const Podcast = getModelForClass(PodcastClass);
```

I'm using this in Next.js 13.4 and I keep getting the following error:

```
Unhandled Runtime Error
TypeError: Right-hand side of 'instanceof' is not an object
src/models/podcasts.ts (179:40) @ PodcastClass
```

How can I fix this?

Thanks in advance!

**Comment by hasezoey** [comment]
do you have a stacktrace and also which version of typegoose you are using?

**Comment by theDanielJLewis** [comment]
Doh! I apologize for neglecting to include that.

Mongoose 7.1.1
Typegoose 11.1.0

Stacktrace from the browser (via `next dev`):

```
Source
mapOptions
node_modules/@typegoose/typegoose/lib/internal/utils.js (382:0)
Object.mapArrayOptions
node_modules/@typegoose/typegoose/lib/internal/utils.js (323:0)
processProp
node_modules/@typegoose/typegoose/lib/internal/processProp.js (289:0)
_buildSchema
node_modules/@typegoose/typegoose/lib/internal/schema.js (35:0)
buildSchema
node_modules/@typegoose/typegoose/lib/typegoose.js (145:0)
processProp
node_modules/@typegoose/typegoose/lib/internal/processProp.js (65:0)
_buildSchema
node_modules/@typegoose/typegoose/lib/internal/schema.js (35:0)
buildSchema
node_modules/@typegoose/typegoose/lib/typegoose.js (145:0)
getModelForClass
node_modules/@typegoose/typegoose/lib/typegoose.js (77:0)
```

The strange thing is the content and even the DB call _does_ load properly, but then it flashes to this error.

**Reply by hasezoey** [reply]
it looks like you have typegoose running in a browser environment, but
typegoose is designed mainly for server side (non browser environments) and is currently made so that if it *loaded* in such a environment to not crash, but using any *functionality* is not supported. the error seems to come from mongoose not having `SchemaTypeOptions` available in the browser build, making it `undefined`, which is invalid for `instanceof`

the general recommendation is one of:
- either dont use mongoose / typegoose in the browser (and use something like graphql), or
- put all typegoose code into a single file (like doing to prevent circular references) (does not affect decorators, but all functions like `buildSchema`, `getModelForClass` and other functions that use mongoose model methods)

**Reply by theDanielJLewis** [reply]
Hmm. I need to understand React Server Components better. I thought this _was_ loading the DB functions server-side instead of in the browser (and they are in a separate file), but maybe I have to make an internal API anyway in order to make this work.
